radix-engine-queries/src/typed_substate_layout.rs

```diff
@@ -167,46 +167,58 @@ pub fn to_typed_object_module_substate_key(
 
 fn to_typed_object_substate_key_internal(
     entity_type: EntityType,
-    module_offset: u8,
+    partition_offset: u8,
     substate_key: &SubstateKey,
 ) -> Result<TypedObjectModuleSubstateKey, ()> {
     let substate_type = match entity_type {
         EntityType::InternalGenericComponent | EntityType::GlobalGenericComponent => {
-            TypedObjectModuleSubstateKey::GenericScryptoComponent(ComponentOffset::try_from(
+            TypedObjectModuleSubstateKey::GenericScryptoComponent(ComponentField::try_from(
                 substate_key,
             )?)
         }
         EntityType::GlobalPackage => {
-            TypedObjectModuleSubstateKey::Package(PackageOffset::try_from(substate_key)?)
+            TypedObjectModuleSubstateKey::Package(PackageField::try_from(substate_key)?)
         }
         EntityType::GlobalFungibleResource => TypedObjectModuleSubstateKey::FungibleResource(
-            FungibleResourceManagerOffset::try_from(substate_key)?,
+            FungibleResourceManagerField::try_from(substate_key)?,
         ),
         EntityType::GlobalNonFungibleResource => {
-            let module_offset = NonFungibleResourceManagerModuleOffset::try_from(module_offset)?;
-            match module_offset {
-                NonFungibleResourceManagerModuleOffset::ResourceManager => {
+            let partition_offset =
+                NonFungibleResourceManagerPartitionOffset::try_from(partition_offset)?;
+            match partition_offset {
+                NonFungibleResourceManagerPartitionOffset::ResourceManager => {
                     TypedObjectModuleSubstateKey::NonFungibleResource(
-                        NonFungibleResourceManagerOffset::try_from(substate_key)?,
+                        NonFungibleResourceManagerField::try_from(substate_key)?,
                     )
                 }
-                NonFungibleResourceManagerModuleOffset::NonFungibleData => {
+                NonFungibleResourceManagerPartitionOffset::NonFungibleData => {
                     let key = substate_key.for_map().ok_or(())?;
                     TypedObjectModuleSubstateKey::GenericKeyValueStore(key.clone())
```

Comment:
For Core API purposes, it would be great if we could instead map this into the well known substate type `TypedObjectModuleSubstateKey::NonFungibleResourceData(x)` and possibly rename the other `TypedObjectModuleSubstateKey::NonFungibleResource` to `NonFungibleResourceField` etc.

Reply:
Yup makes sense!